### PR TITLE
New: Unify series custom filter options

### PR DESCRIPTION
**Pull request by RobinDadswell**
#### Database Migration
YES 151

#### Description
Unification of options available for use within custom filter options across Series Index, Mass Editor and Season Pass

#### Issues Fixed or Closed by this PR

* #3548 

Currently an issue with build as values are assigned but not used to enable filtering to work.

### CODE REVIEW

**Comment by RobinDadswell** [comment]
Build issue has been resolved; this is now ready for feedback

**Comment by RobinDadswell** [comment]
Just got to make custom filters viewable across all 3 pages left to do.

**Comment by RobinDadswell** [comment]
@markus101  this is ready for review, let me know if there is anything you want changed (DB migration is 151 based on the other 2 PR's with DB migrations (mine at 149 and Taloth with one at 150)

File: frontend/src/Store/Actions/seasonPassActions.js [hunk 1]
```diff
@@ -25,7 +26,26 @@ export const defaultState = {
   secondarySortDirection: sortDirections.ASCENDING,
   selectedFilterKey: 'all',
   filters,
-  filterPredicates,
+  filterPredicates: {
+    ...filterPredicates,
+
+    episodeProgress: function(item, filterValue, type) {
```
Comment (markus101) [inline frontend/src/Store/Actions/seasonPassActions.js:32]: This should be added to `seriesActions.filterPredicates` and removed from the 3 action files.

File: frontend/src/Store/Actions/seasonPassActions.js [hunk 2]
```diff
@@ -42,8 +62,28 @@ export const defaultState = {
     },
     {
       name: 'seriesType',
-      label: 'Series Type',
-      type: filterBuilderTypes.EXACT
+      label: 'Type',
```
Comment (markus101) [inline frontend/src/Store/Actions/seasonPassActions.js:65]: Let's move all the common filter builder props to `seriesActions` and remove the dupes from the 3 other action files.

**Review by markus101 — CHANGES_REQUESTED** [review]
The big thing is de-duplicating the copy pasta from the 3 action files and moving it to `seriesActions` so it can be shared. That'll cut down on the code and make it easy to fix all 3 at once if we break something.

File: frontend/src/Store/Actions/seasonPassActions.js (hunk 2 above)
Comment (RobinDadswell) [reply frontend/src/Store/Actions/seasonPassActions.js]: Done

File: frontend/src/Store/Actions/seasonPassActions.js (hunk 1 above)
Comment (RobinDadswell) [reply frontend/src/Store/Actions/seasonPassActions.js]: Done

**Comment by RobinDadswell** [comment]
@markus101  ready once again :)

**Comment by markus101** [comment]
Thanks @RobinDadswell!

**Pull request merged**